src/breakpoints.jl

```diff
@@ -7,6 +7,37 @@ Return an array with all breakpoints.
 """
 breakpoints() = _breakpoints
 
+
+const breakpoint_update_hooks = []
+"""
+    on_breakpoints_updated(f)
+
+Register a one-argument function to be called after any update to the global set of
+breakpoints. This includes their creation, deletion, enabling and disabling.
+
+The function `f` should take two inputs:
+First argument is the function doing to update, such as `::typeof(breakpoint)` for the
+creation, or `::typeof(remove)` for the deletion.
+Second argument is the breakpoint object that was changed.
+If only desiring to handle some kinds of update, `f` should have fallback methods
+to do nothing in the general case.
+
+!!! warning
+    This feature is experimental, and may be modified or removed without the tagging of a
+    breaking release.
+"""
+on_breakpoints_updated(f) = push!(breakpoint_update_hooks, f)
+
+function firehooks(hooked_fun, bp::AbstractBreakpoint)
+    for hook in breakpoint_update_hooks
+        try
+            hook(hooked_fun, bp)
```

Comment:
Does this need `invokelatest`?

Reply:
I would not expect the hook functions to be redefined during the execution of the program.
Juno of MixedModeDebugger would set them at the start and then done.

So I don't think it needs to be.

Reply:
But it can be, if you think its better